modules/nw-http2-haproxy.adoc

```diff
@@ -0,0 +1,46 @@
+// Module included in the following assemblies:
+//
+// * networking/ingress-operator.adoc
+
+[id="nw-http2-haproxy{context}"]
+= HTTP/2 Ingress connectivity
+
+You can enable transparent end-to-end HTTP/2 connectivity in HAProxy. It allows application owners to make use of HTTP/2 protocol capabilities, including single connection, header compression, binary streams, and more.
+
+You can enable this HTTP/2 connectivity for an individual Ingress Controller or for the entire cluster.
+
+To enable the use of HTTP/2 for the connection from the client to HAProxy, a
+route must specify a custom certificate. A route that uses the default
+certificate cannot use HTTP/2. This restriction is necessary to avoid problems
+from connection coalescing, where the client re-uses a connection for different
+routes that use the same certificate.
+
+The connection from HAProxy to the application Pod can use HTTP/2 only for
+re-encrypt routes and not for edge-terminated or insecure routes. This
+restriction comes from the fact that HAProxy uses Application-Level Protocol
+Negotiation (ALPN), which is a TLS extension, to negotiate the use of HTTP/2
+with the back-end. The implication is that end-to-end HTTP/2 is possible with
+passthrough and re-encrypt and not with insecure or edge-terminated routes.
+
+[IMPORTANT]
+====
+A connection that uses the HTTP/2 protocol cannot be upgraded to the WebSocket protocol. If you have a back-end application that is designed to allow WebSocket connections, it must not allow a connection to negotiate use of the HTTP/2 protocol or else WebSocket connections will fail.
+====
+
+.Procedure
+
+Enable HTTP/2 on a single Ingress Controller.
+
+* To enable HTTP/2 on an Ingress Controller, run the `oc annotate` command:
```

Comment:
I've been using _enter_ instead of _run_.

> ISG: To instruct users to enter commands, use type, enter, or issue. Use one term consistently through- out your information. To refer to the area where you enter commands, in documentation for only Windows operating systems, use command prompt. In documentation for other operating sys- tems or for both Windows and other operating systems, use command line. Do not use run to refer to specifying a command on a command line or at a command prompt. However, you can use run to refer to the general use of a command. (p. 186)